mllam_data_prep/ops/derive_variable/main.py

```diff
@@ -0,0 +1,259 @@
+                # set during the calculation of the derived variable
+                raise KeyError(
+                    f'The attribute "{attribute}" has not been set for the derived'
+                    f' variable "{field.name}". This is most likely because you are'
+                    " using a function external to `mlllam-data-prep` to derive the field,"
+                    f" in which the required attributes ({', '.join(REQUIRED_FIELD_ATTRIBUTES)})"
+                    " are not set. If they are not set in the function call when deriving the field,"
+                    ' they can be set in the config file by adding an "attrs" section under the'
+                    f' "{field.name}" derived variable section. For example, if the required attributes'
+                    f" ({', '.join(REQUIRED_FIELD_ATTRIBUTES)}) are not set for a derived variable named"
+                    f' "toa_radiation" they can be set by adding the following to the config file:'
+                    ' {"attrs": {"units": "W*m**-2", "long_name": "top-of-atmosphere incoming radiation"}}.'
+                )
+        elif attribute in expected_attributes.keys():
+            logger.warning(
+                f"The attribute '{attribute}' of the derived field"
+                f" {field.name} is being overwritten from"
+                f" '{field.attrs[attribute]}' to"
+                f" '{expected_attributes[attribute]}' according"
+                " to the specification in the config file."
+            )
+            attrs[attribute] = expected_attributes[attribute]
+        else:
+            # Attributes are set in the function and nothing has been defined in the config file
+            attrs[attribute] = field.attrs[attribute]
+
+    return attrs
+
+
+def _align_derived_variable(field, ds, target_dims):
+    """
+    Align a derived variable to the target dimensions (ignoring non-dimension coordinates).
+
+    Parameters
+    ----------
+    field: xr.DataArray
+        Derived field to align
+    ds: xr.Dataset
+        Target dataset
+    target_dims: List[str]
+        Dimensions to align to (e.g. 'time', 'y', 'x')
+
+    Returns
+    -------
+    field: xr.DataArray
+        The derived field aligned to the target dimensions
+    """
+    # Ensure that dimensions are ordered correctly
+    field = field.transpose(
+        *[dim for dim in target_dims if dim in field.dims], missing_dims="ignore"
+    )
+
+    # Add missing dimensions explicitly
+    for dim in target_dims:
+        if dim not in field.dims:
+            field = field.expand_dims({dim: ds.sizes[dim]})
+
+    # Broadcast to match only the target dimensions
+    broadcast_shape = {dim: ds[dim] for dim in target_dims if dim in ds.dims}
+    field = field.broadcast_like(xr.Dataset(coords=broadcast_shape))
+
+    return field
+
+
+def get_latlon_coords_for_input(ds):
+    """Dummy function for getting lat and lon."""
```

Comment:
Maybe we should instead write:

```suggestion
    """Placeholder function for getting latitude and longitude values. This will eventually be 
    replaced by routines handling proper projection support
    (see https://github.com/mllam/mllam-data-prep/pull/38)
    """
```